lib/oanda_api_v20/pricing.rb

```diff
@@ -8,26 +8,29 @@ def pricing(options)
 
     # GET /v3/accounts/:account_id/pricing/stream
     def pricing_stream(options, &block)
-      buffer = StringIO.new
+      buffer = String.new
 
       Client.send(http_verb, "#{base_uri}/accounts/#{account_id}/pricing/stream", headers: headers, query: options, stream_body: true) do |fragment|
-        begin
-          next if fragment.empty?
-
+        if !fragment.empty?
           buffer << fragment
-          next unless fragment.match(/\n\Z/)
-
-          buffer.string.split("\n").each do |message|
-            cleaned_message = message.strip
-            next if cleaned_message.empty?
-            yield JSON.parse(cleaned_message)
-          end
-        rescue JSON::ParseError => e
-          raise OandaApiV20::ParseError, "#{e.message} in '#{fragment}'"
-        ensure
-          buffer.flush
+          parse(buffer, fragment, &block) if fragment.match(/\n\Z/)
         end
       end
     end
+
+    private
+
+    def parse(buffer, fragment, &block)
+      buffer.split("\n").each do |message|
+        cleaned_message = message.strip
+        next if cleaned_message.empty?
+        yield JSON.parse(cleaned_message)
+      end
+    rescue JSON::ParserError => e
+      raise OandaApiV20::ParseError, "#{e.message} in '#{fragment}'"
+    ensure
+      buffer.clear
+    end
+
```

Comment:
Can we maybe just remove this empty line here. Amazing work, thank you so much. The rest is perfect!